shiki/WordPress-Android
# Import from Giphy in Post Editor

![dec-06-2018 07-15-10](https://user-images.githubusercontent.com/198826/49589463-b5dc7e80-f926-11e8-96ec-aa092484bb27.gif)

This is a part of the Giphy picker feature (wordpress-mobile/WordPress-Android#8627). It adds importing of GIFs from Giphy from the Post Editor.

The changes involve reusing the existing `EditPostActivity.startUploadService` to upload the local `MediaModel` received from the Giphy Picker. After that, `EditorFragment.appendMediaFile` is used to add them to the editor. The other required changes are already handled in #3.

The `develop` branch from [wordpress-mobile/WordPress-Android](/wordpress-mobile/WordPress-Android/tree/develop) was merged before the changes. Merge commit is f95b953ff9. The first commit for implementing this feature is 831fc396a2.

### Testing

1. Create a [Giphy API key](https://developers.giphy.com/docs/) and paste it in your `gradle.properties` as the value for `wp.giphy.api_key`
2. Create or edit a post
3. Click on the plus (+) button at the bottom left. Choose the photo icon and then the _Choose from Giphy_ option.
4. In the Giphy picker, search and pick some GIFs
5. Press the _Add_ button. The selected GIFs should be downloaded and the picker should be closed. The images will be added in the Editor and the upload will commence. The GIF icon will show up in the images once they finish downloading.
